keras/layers/recurrent.py

```diff
@@ -570,6 +577,8 @@ def __call__(self, inputs, initial_state=None, constants=None, **kwargs):
             self.input_spec = full_input_spec
             output = super(RNN, self).__call__(full_input, **kwargs)
             self.input_spec = original_input_spec
+            for arg in args_to_pop:
```

Comment:
You shouldn't need to pop these args here. Rather, you should remove them from kwargs before calling the layer (which is fine since they are transferred to the input list)